src/astro/files/base.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import create_file_location
+from astro.files.type import create_type_factory
+
+
+class File:
```

Comment:
@utkarsharma2 I think it would be great to add a docstring explaining if we expect the `File` to be user-facing.

Reply:
@tatiana, 
1. I'll the docstring.
2. @dimberman had a different view on moving this in __init__.py. File was initially in __init__.py then moved it to base.py.

Also, the API I want for this module is to just expose the Class File, which then can be used by client code, and not end user. 

If we move File class to __init__.py, there we need to import FileType, create_file_location and create_type_factory, which will the be exposed in the namespace, which I don't want the client code to use directly as it break the intended API.

Reply:
@utkarsharma2 by "make it available" I meant to allow users to import it using:
```
from astro.files import File
```

This can be achieved by simply

Reply:
It's already added.